lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -9,24 +10,34 @@ class PokemonOverviewPage extends StatelessWidget {
     Key? key,
   }) : super(key: key);
 
-  final List<Pokemon> pokemons;
+  final Async<List<Pokemon>> pokemons;
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(title: const Text(pokedexTitle)),
-      body: GridView.builder(
-        gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
-          crossAxisCount: 2,
-          childAspectRatio: 1,
-          crossAxisSpacing: 10,
-          mainAxisSpacing: 10,
+      body: pokemons.when(
+        loading: () => const Center(child: CircularProgressIndicator()),
+        error: (String? errorMessage) => Center(
+          child: Text(
+            errorMessage!,

```

Comment:
instead of enforcing, use null coalescing here then pass an empty string instead so it's safer 🙂

Reply:
resolved